sympy/solvers/ode.py

```diff
@@ -1482,6 +1491,233 @@ def _test_term(coeff, order):
+
+    A = cancel((match['r'][b3])/((match['r'])[a3]))
+    B = cancel((match['r'][c3])/((match['r'])[a3]))
+
+    sol = None
+    if match['type'] == "2F1":
+        if c.is_integer == False or c.is_positive is not True:
+            sol = C0*hyper([a, b], [c], x) + C1*hyper([a-c+1, b-c+1], [2-c], x)*x**(1-c)
+        elif c == 1:
+            y2 = Integral(exp(Integral((-(a+b+1)*x + c)/(x**2-x), x))/hyperexpand(hyper([a, b], [c], x)**2), x)*hyper([a, b], [c], x)
+            sol = C0*hyper([a, b], [c], x) + C1*y2
+        # applying transformation in the solution
+        subs = match['mobius']
+        dtdx = simplify(1/(subs.diff(x)))
+        _B = simplify(((a + b + 1)*x - c).subs(x, subs)*dtdx)
+        _B = simplify(_B + ((x**2 -x).subs(x, subs))*(dtdx.diff(x)*dtdx))
+        _A = simplify((x**2 - x).subs(x, subs)*(dtdx**2))
+        e = exp(logcombine(Integral(cancel(_B/(2*_A)), x).doit(), force=True))
+        sol = (sol).subs(x, match['mobius'])
+        sol = sol.subs(x, x**match['k'])
+        e = (e).subs(x, x**match['k'])
+
+        if B:
+            e1 = Integral(A/2, x).doit()
+            e1 = exp(logcombine(e1, force=True))
+
+        sol = cancel(powdenest(cancel(e/e1)*x**((-match['k']+1)/2), force=True))*(sol)
+        sol = Eq(func, sol)
+
+    # if sol is None then we can try for series solution
+    if sol is None:
+        try:
+            sol = dsolve(eq, func, hint = "2nd_power_series_regular", x0=x0)
+        except ValueError:
+            pass
+    if sol is None:
+        raise NotImplementedError("....")
+
+    return sol
+
+def match_2nd_2F1_hypergeometric(I, k, sing_point, func, r):
+
+    from sympy import factor
+    x = func.args[0]
+    a = Wild("a")
+    b = Wild("b")
+    c = Wild("c")
+    t = Wild("t")
+    alpha = Wild("alpha")
+    beta = Wild("beta")
+    gamma = Wild("gamma")
+    delta = Wild("delta")
+
+    rn = {'type':None}
+    # I0 of the standerd 2F1 equation.
+    I0 = ((a-b+1)*(a-b-1)*x**2 + 2*((1-a-b)*c + 2*a*b)*x + c*(c-2))/(4*x**2*(x-1)**2)
+    if sing_point != [0, 1]:
+        # If singular point is [0, 1] then we have standerd equation.
+        eqs = []
+        sing_eqs = [-beta/alpha, -delta/gamma, (delta-beta)/(alpha-gamma)]
+        # making equations for the finding the mobius transformation
+        for i in range(3):
+            if i<len(sing_point):
+                eqs.append(Eq(sing_eqs[i], sing_point[i]))
+            else:
+                eqs.append(Eq(1/sing_eqs[i], 0))
+        # solving above equations for the mobius transformation
+        _beta = solve(eqs[0], beta)[0]
+        _delta = solve(eqs[1], delta)[0]
+        _gamma = solve(((eqs[2]).subs(beta, _beta)).subs(delta, _delta), gamma)[0]
+        subs = (alpha*x + beta)/(gamma*x + delta)
+        subs = subs.subs(beta, _beta)
+        subs = subs.subs(delta, _delta)
+        subs = subs.subs(gamma, _gamma)
+        subs = cancel(subs)
+    else:
+        subs = x
+    t = (solve(Eq(subs, t), x)[0]).subs(t, x)
```

Comment:
Will `solve` only return one solution?

Reply:
yes

Reply:
Is that because this is a polynomial? If so can we use roots?